src/transaction.js

```diff
@@ -120,14 +120,12 @@ Transaction.fromBuffer = function (buffer, __noStrict) {
   }
 
   if (hasWitnesses) {
-    var isNull = true
     for (i = 0; i < vinLen; ++i) {
       tx.ins[i].witness = readVector()
-      isNull = isNull && tx.ins[i].witness.length === 0
-    }
-    if (isNull) {
-      throw new Error('Transaction has superfluous witness data')
     }
+
+    // was this pointless?
+    if (!tx._hasWitnesses()) throw new Error('Transaction has superfluous witness data')
```

Comment:
Since it addresses the only erroneous conversion between base & witness formats I'm not sure it is. Our test cases were happily accepting transactions that Bitcoin Core wouldn't, this check would have prevented that.